haskellPackages: Stackage 23.17 -> 23.19

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the \`haskell-updates\` branch into \`staging\`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these \`haskell-updates\` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [\`pkgs/development/haskell-modules/HACKING.md\`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on \`haskell-updates\` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of \`haskell-updates\` into \`staging\` every two weeks.
* We only do the merge if the [\`mergeable\`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [\`maintained\`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #371032. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

wolfgangwalther committed:
haskell.packages.ghc9122.interpolate: disable tests due to GHC bug

wolfgangwalther committed:
haskell.packages.ghc9122.hpack: 0.37.0 -> 0.38.0

Comes with a jailbreak for Cabal 3.14.

sternenseemann committed:
haskellPackages: stackage LTS 23.17 -> LTS 23.19

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
all-cabal-hashes: 2025-03-30T11:13:14Z -> 2025-04-21T04:38:52Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages: Cabal* 3.14.1.* -> 3.14.2.0

sternenseemann committed:
haskellPackages.network-control: use version from stackage

which now matches the version we want!

nixpkgs-ci[bot] committed:
Merge 29662380daddf1bdf7cc3cf48c4fed04e574602d into haskell-updates

sternenseemann committed:
stack: pin hpack to == 0.38.0 for 3.5.1

sternenseemann committed:
git-annex: update sha256 for 10.20250416

wolfgangwalther committed:
haskellPackages.cabal2nix-unstable: 2025-04-22 -> 2025-04-23

wolfgangwalther committed:
postgrest: 12.2.7 -> 12.2.11

Release notes:
https://github.com/PostgREST/postgrest/releases/tag/v12.2.11
https://github.com/PostgREST/postgrest/releases/tag/v12.2.10
https://github.com/PostgREST/postgrest/releases/tag/v12.2.9
https://github.com/PostgREST/postgrest/releases/tag/v12.2.8

maralorn commented:
Maybe those are genuine conflicts between master and staging? But normally the merge job should prevent those from reaching our branch, right?

maralorn committed:
haskell.packages.ghc9122.cabal2nix: fix build (#400177)

maralorn committed:
postgrest: 12.2.7 -> 12.2.11 (#401175)

nixpkgs-ci[bot] committed:
Merge 7c3df50c30ec6dd4fb6e26411cb4cd449863f4ba into haskell-updates

nixpkgs-ci[bot] committed:
Merge 9979a56ae65660cc2585e0c70d3252979e89ae44 into haskell-updates

sternenseemann commented:
We have done this a few times before. This sort of thing does happen. Unfortunately, the nixpkgs CI has not been designed with merge conflicts in mind which not only affects us, but is also a nuisance when dealing with staging(-next). The only CI that really matters, though, is Hydra.

We are free to cherry pick commits back and forth which is probably wise, so that interactions between changes are properly tested.

dragospe committed:
haskellPackages.accelerate: overrideSrc from GH

The last official release of accelerate appears to have been 5 years
ago and does not build with our GHC. However, the repository is active
 and appears at least to build up to GHC 9.10.

This PR pulls the source from github and adds missing
dependencies.

marcin-serwin committed:
haskellPackages.hail: unbreak

wolfgangwalther committed:
haskellPackages.aeson-better-errors: fix build

wolfgangwalther committed:
haskellPackages.mvc-updates: fix build

wolfgangwalther committed:
haskellPackages.taskwarrior: fix build

wolfgangwalther committed:
haskell.packages.ghc912.generic-arbitrary: fix build

nixpkgs-ci[bot] committed:
Merge 2169c869932137ec6173c7fc892982778c37c548 into haskell-updates

wolfgangwalther committed:
haskellPackages: remove non-applying patches (#401839)

wolfgangwalther committed:
haskellPackages.hail: unbreak (#401830)

wolfgangwalther committed:
haskellPackages.accelerate: overrideCabal from GH (#401552)

wolfgangwalther committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

nixpkgs-ci[bot] committed:
Merge 0571e373ab39b6dd0564ac94b3b341f43dc4f4d7 into haskell-updates

alexfmpe committed:
haskellPackages.reflex-dom: remove eval warning

nixpkgs-ci[bot] committed:
Merge 917ee0af83188b95e3d6f844f10e1b33acd0bcdb into haskell-updates

nixpkgs-ci[bot] committed:
Merge 3701dbcbc421a382eb884506120fc2240b026ee4 into haskell-updates

sternenseemann committed:
Merge commit 72d752137489 into haskell-updates

sternenseemann committed:
haskellPackages.what4_1_7: also apply overrides

sternenseemann committed:
haskellPackages.crucible*: use what4 >= 1.7

Explicitly requested by crucible-llvm.

sternenseemann committed:
haskellPackages.crucible-llvm: provide test dependency z3

sternenseemann committed:
haskellPackages.crux: provide simple-get-opt < 0.5

sternenseemann committed:
haskellPackages.sensei: disable test suite

sternenseemann committed:
haskellPackages.sensei: drop outdated override

sternenseemann committed:
haskellPackages.reflex-dom: reformat override

Commit:
haskellPackages.cabal2nix-unstable: 2025-04-23 -> 2025-04-30

Adds support for the pkg-config python3-embed library name.

sternenseemann committed:
haskellPackages.pinecone: disable network dependent tests

sternenseemann committed:
haskellPackages.jsonpatch: disable broken test suite

wolfgangwalther committed:
haskellPackages.postgresql-libpq-configure: unbreak

We split pg_config from libpq, so need to provide it manually for those
packages that can't make use of pkgconfig.

wolfgangwalther committed:
haskellPackages.HDBC-postgresql: unbreak

We split pg_config from libpq a while ago, so need to provide it
manually for those packages that can't use pkgconfig.

maralorn committed:
haskellPackages.{postgresql-libpq-configure,HDBC-postgresql}: unbreak (#403431)

nixpkgs-ci[bot] committed:
Merge b5dd9efc912ecabeafa4f082d31e19cb1c74266c into haskell-updates

wolfgangwalther committed:
haskellPackages: unbreak various packages

This is the result of running the unbreak.nu script.

wolfgangwalther committed:
haskellPackages.postgrest: 12.2.11 -> 12.2.12

Release notes:
https://github.com/PostgREST/postgrest/releases/tag/v12.2.12

wolfgangwalther committed:
haskellPackages: misc fixes (#403419)

wolfgangwalther committed:
haskellPackages.postgrest: 12.2.11 -> 12.2.12 (#403553)

wolfgangwalther committed:
[haskell-updates] haskellPackages.crux: unbreak (#403343)

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1814999](https://hydra.nixos.org/eval/1814999) of nixpkgs commit [3937cbe](https://github.com/NixOS/nixpkgs/commits/3937cbe4d3846be40349599504cef1765d8e4d59) as of 2025-05-02 08:45 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann committed:
Merge commit 97e6d418edc7 into haskell-updates

sternenseemann commented:
Ready to merge, I think, anything else?

sternenseemann commented:
Shouldn't that bug be fixed by now?
